spec/services/charges/charge_models/graduated_service_spec.rb

```diff
@@ -48,48 +48,176 @@
+    it 'returns expected amount', :aggregate_failures do
       expect(apply_graduated_service.amount).to eq(12)
+      expect(apply_graduated_service.unit_amount).to eq(12)
+      expect(apply_graduated_service.amount_details).to eq(
+        {
+          graduated_ranges: [
+            {
+              flat_amount: 2,
+              from_value: 0,
+              to_value: 10,
+              total_amount: 10,
+              total_with_flat_amount: 12,
+              unit_amount: 10,
+              units: 1,
+            },
+          ],
+        },
+      )
     end
   end
 
   context 'when aggregation is 10' do
     let(:aggregation) { 10 }
 
-    it 'applies all unit amount for top bound' do
+    it 'returns expected amount', :aggregate_failures do
       expect(apply_graduated_service.amount).to eq(102)
+      expect(apply_graduated_service.unit_amount).to eq(10.2)
+      expect(apply_graduated_service.amount_details).to eq(
+        {
+          graduated_ranges: [
+            {
+              flat_amount: 2,
+              from_value: 0,
+              to_value: 10,
+              total_amount: 100,
+              total_with_flat_amount: 102,
+              unit_amount: 10,
+              units: 10,
+            },
+          ],
+        },
+      )
     end
   end
 
   context 'when aggregation is 11' do
     let(:aggregation) { 11 }
 
-    it 'applies next range flat amount for the next step' do
+    it 'returns expected amount', :aggregate_failures do
       expect(apply_graduated_service.amount).to eq(110)
+      expect(apply_graduated_service.unit_amount).to eq(10)
+      expect(apply_graduated_service.amount_details).to eq(
+        {
+          graduated_ranges: [
+            {
+              flat_amount: 2,
+              from_value: 0,
+              to_value: 10,
+              total_amount: 100,
+              total_with_flat_amount: 102,
+              unit_amount: 10,
+              units: 10,
+            }, {
+              flat_amount: 3,
+              from_value: 11,
+              to_value: 20,
+              total_amount: 5,
+              total_with_flat_amount: 8,
+              unit_amount: 5,
+              units: 1,
+            },
+          ],
+        },
+      )
     end
   end
 
   context 'when aggregation is 12' do
     let(:aggregation) { 12 }
 
-    it 'applies next unit amount for more unit in next step' do
+    it 'returns expected amount', :aggregate_failures do
       expect(apply_graduated_service.amount).to eq(115)
+      expect(apply_graduated_service.unit_amount.round(5)).to eq(9.58333)
+      expect(apply_graduated_service.amount_details).to eq(
+        {
+          graduated_ranges: [
+            {
+              flat_amount: 2,
+              from_value: 0,
+              to_value: 10,
+              total_amount: 100,
+              total_with_flat_amount: 102,
+              unit_amount: 10,
+              units: 10,
+            }, {
```

Comment:
Layout/MultilineArrayLineBreaks: Each item in a multi-line array must start on a separate line.